src/sentry/tasks/collect_project_platforms.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import
+
+from datetime import timedelta
+from django.utils import timezone
+
+from sentry.constants import VALID_PLATFORMS
+from sentry.models import Group, ProjectPlatform
+from sentry.tasks.base import instrumented_task
+
+
+@instrumented_task(name='sentry.tasks.collect_project_platforms', queue='stats')
+def collect_project_platforms(**kwargs):
+    now = timezone.now()
+    queryset = Group.objects.filter(
+        last_seen__gte=now - timedelta(days=1),
+        platform__isnull=False,
+    ).values_list('platform', 'project_id').distinct()
+
+    for platform, project_id in queryset:
```

Comment:
Yeah, what about just breaking it up into smaller chunks? And say, operating on 24 hour buckets, instead of the entire 24 hours in one big query? Or some reasonable chunk size.